base/show.jl

```diff
@@ -326,7 +326,10 @@ show(x) = show(stdout::IO, x)
 
 show(io::IO, @nospecialize(x)) = show_default(io, x)
 
-function show_default(io::IO, @nospecialize(x))
+# avoid inferring show_default on the type of `x`
+show_default(io::IO, @nospecialize(x)) = _show_default(io, Ref{Any}(x)[])
```

Comment:
I'd rather add an explicit `inferencebarrier` function (can have this implementation for now), just so we can keep that working if inference ever learns to deal with this.

Reply:
Ok, good idea.